persistence/trees/trees_test.go

```diff
@@ -1,19 +1,225 @@
+
+	DefaultStakeStatus     = int32(coreTypes.StakeStatus_Staked)
+	DefaultPauseHeight     = int64(-1) // pauseHeight=-1 implies not paused
+	DefaultUnstakingHeight = int64(-1) // unstakingHeight=-1 implies not unstaking
+
+	testSchema = "test_schema"
+
+	genesisStateNumValidators   = 5
+	genesisStateNumServicers    = 1
+	genesisStateNumApplications = 1
+)
 
-// TECHDEBT(#836): Tests added in https://github.com/pokt-network/pocket/pull/836
 func TestTreeStore_Update(t *testing.T) {
-	// TODO: Write test case for the Update method
-	t.Skip("TODO: Write test case for Update method")
+	pool, resource, dbUrl := test_artifacts.SetupPostgresDocker()
+	t.Cleanup(func() {
+		require.NoError(t, pool.Purge(resource))
+	})
+
+	t.Run("should update actor tree and commit", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := NewTestPostgresContext(t, 0, pmod)
+
+		require.NoError(t, context.NewSavePoint())
+
+		hash1, err := context.ComputeStateHash()
+		require.NoError(t, err)
+		require.NotEmpty(t, hash1)
+
+		_, err = createAndInsertDefaultTestApp(t, context)
+		require.NoError(t, err)
+
+		require.NoError(t, context.NewSavePoint())
+
+		hash2, err := context.ComputeStateHash()
+		require.NoError(t, err)
+		require.NotEmpty(t, hash2)
+		require.NotEqual(t, hash1, hash2)
+	})
+}
+
+func newTestPersistenceModule(t *testing.T, databaseUrl string) modules.PersistenceModule {
+	t.Helper()
+	teardownDeterministicKeygen := keygen.GetInstance().SetSeed(42)
+	defer teardownDeterministicKeygen()
+
+	cfg := &configs.Config{
+		Persistence: &configs.PersistenceConfig{
+			PostgresUrl:       databaseUrl,
+			NodeSchema:        testSchema,
+			BlockStorePath:    ":memory:",
+			TxIndexerPath:     ":memory:",
+			TreesStoreDir:     ":memory:",
+			MaxConnsCount:     5,
+			MinConnsCount:     1,
+			MaxConnLifetime:   "5m",
+			MaxConnIdleTime:   "1m",
+			HealthCheckPeriod: "30s",
+		},
+	}
+
+	genesisState, _ := test_artifacts.NewGenesisState(
+		genesisStateNumValidators,
+		genesisStateNumServicers,
+		genesisStateNumApplications,
+		genesisStateNumServicers,
+	)
+
+	runtimeMgr := runtime.NewManager(cfg, genesisState)
+
+	bus, err := runtime.CreateBus(runtimeMgr)
+	require.NoError(t, err)
+
+	persistenceMod, err := persistence.Create(bus)
+	require.NoError(t, err)
+
+	return persistenceMod.(modules.PersistenceModule)
+}
+func createAndInsertDefaultTestApp(t *testing.T, db *persistence.PostgresContext) (*coreTypes.Actor, error) {
+	t.Helper()
+	app := newTestApp(t)
+
+	addrBz, err := hex.DecodeString(app.Address)
+	require.NoError(t, err)
+
+	pubKeyBz, err := hex.DecodeString(app.PublicKey)
+	require.NoError(t, err)
+
+	outputBz, err := hex.DecodeString(app.Output)
+	require.NoError(t, err)
+	return app, db.InsertApp(
+		addrBz,
+		pubKeyBz,
+		outputBz,
+		false,
+		DefaultStakeStatus,
+		DefaultStake,
+		DefaultChains,
+		DefaultPauseHeight,
+		DefaultUnstakingHeight)
+}
+
+func newTestApp(t *testing.T) *coreTypes.Actor {
```

Comment:
Can you leave a few TECHDEBT comments linking to the ticket you and @bryanchriswhite created relating to consolidating testing helpers?